Remotion/Data/DomainObjects.UnitTests/Infrastructure/DomainObjectTransactionContextIndexerTest.cs

```diff
@@ -29,24 +30,13 @@ public void Item ()
     {
       var order = DomainObjectIDs.Order1.GetObject<Order>();
       var tx = order.RootTransaction.CreateSubTransaction();
-      var indexer = new DomainObjectTransactionContextIndexer(order, false);
+      var indexer = new DomainObjectTransactionContextIndexer(new DomainObjectTransactionContextImplementation(order));
 
       var item = indexer[tx];
-      Assert.That(item, Is.InstanceOf(typeof(DomainObjectTransactionContext)));
-      Assert.That(((DomainObjectTransactionContext)item).DomainObject, Is.SameAs(order));
-      Assert.That(item.ClientTransaction, Is.SameAs(tx));
-    }
-
-    [Test]
-    public void Item_WhileEventIsExecuting ()
-    {
-      var order = DomainObjectIDs.Order1.GetObject<Order>();
-      var tx = order.RootTransaction.CreateSubTransaction();
-      var indexer = new DomainObjectTransactionContextIndexer(order, true);
 
-      var item = indexer[tx];
-      Assert.That(item, Is.InstanceOf(typeof(InitializedEventDomainObjectTransactionContextDecorator)));
       Assert.That(item.ClientTransaction, Is.SameAs(tx));
+      var transactionContextImplementation = PrivateInvoke.GetNonPublicField(item, "_transactionContextImplementation");
+      Assert.That(transactionContextImplementation, Is.Not.Null);
```

Comment:
Step 2:
```suggestion
      Assert.That(item, Is.EqualTo(new DomainObjectTranscationContext(transactionContextImplementation, tx));
```

This works because DOTxCtx is a value type so there is implicit state equality.